app/controllers/Extractors.scala

```diff
@@ -80,6 +112,89 @@ class Extractors  @Inject() (extractions: ExtractionService,
     }
   }
 
+  def showExtractorMetrics(extractorName: String) = AuthenticatedAction { implicit request =>
+    implicit val user = request.user
+
+    val dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS")
+    val todaydate = dateFormatter.format(new java.util.Date())
+
+    val hourcal = Calendar.getInstance
+    hourcal.add(Calendar.HOUR, -24)
+    val yesterdaydate = dateFormatter.format(hourcal.getTime)
+
+    val daycal = Calendar.getInstance
+    daycal.add(Calendar.WEEK_OF_MONTH, -1)
+    val last7daydate = dateFormatter.format(daycal.getTime)
+
+    val monthcal = Calendar.getInstance
+    monthcal.add(Calendar.MONTH, -1)
+    val lastmonthdate = dateFormatter.format(monthcal.getTime)
+
+    Logger.debug("today date: " + todaydate)
+    Logger.debug("yesterday date: " + yesterdaydate)
+    Logger.debug("last 7 day date: " + last7daydate)
+    Logger.debug("last month date: " + lastmonthdate)
+
+    // get at most 10 ``DONE'' submission from yesterday.
+    val myDonelist = extractions.findByExtractorIDBefore(extractorName, "DONE", yesterdaydate, 10)
+    // get at most 10 ``SUBMITTED'' submission from yesterday.
+    val mySubmittedlist = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", yesterdaydate, 10)
+    var n: Long = 0
+    var sum: Long = 0
+    // calculate the diff time between ``SUBMITTED'' and ``DONE''.
+    for ((done, submitted) <- (myDonelist zip mySubmittedlist)) {
+      Logger.debug("my done date: " + done.start + ", fileid: " + done.file_id)
+      Logger.debug("my submitted date: " + submitted.start + ", fileid: " + submitted.file_id)
+      Logger.debug("done.start.getTime: " + done.start.getTime + ", submitted.start.getTime: " + submitted.start.getTime)
+      val diffInMillies = Math.abs(done.start.getTime - submitted.start.getTime)
+      Logger.debug("diffInMillies in ms: " + diffInMillies)
+      sum = sum + diffInMillies
+      n = n+1
+    }
+    sum = TimeUnit.SECONDS.convert(sum, TimeUnit.MILLISECONDS)
+    var average = BigDecimal.valueOf(sum)
+    Logger.debug("average: " + average)
+    if(n > 0) {
+      average = average/n
+    }
+    Logger.debug("average: " + average)
+    // get at the number of ``SUBMITTED'' submission in last week.
+    val lastweeksubmitted = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", last7daydate, 0)
+    Logger.debug("lastweek submitted: " + lastweeksubmitted.size)
+    // get at the number of ``SUBMITTED'' submission in last month.
+    val lastmonthsubmitted = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", lastmonthdate, 0)
+
```

Comment:
This should count the number of unique jobs submitted, instead of the number of raw events we've seen:
```suggestion
    Logger.debug("average: " + average)
    // get at the number of ``SUBMITTED'' submission in last week.
    val lastweekevents = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", last7daydate, 0)
    val lastweeksubmitted = lastweekevents.map(evt => evt.job_id).distinct.length
    Logger.debug("lastweek submitted: " + lastweeksubmitted)
    // get at the number of ``SUBMITTED'' submission in last month.
    val lastmonthevents = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", lastmonthdate, 0)
    val lastmonthsubmitted = lastmonthevents.map(evt => evt.job_id).distinct.length

```